packages/contracts/contracts/TransactionManager.sol

```diff
@@ -505,7 +511,18 @@ contract TransactionManager is ReentrancyGuard, ProposedOwnable, ITransactionMan
     }
 
     // Emit event
-    emit TransactionFulfilled(txData.user, txData.router, txData.transactionId, txData, relayerFee, signature, callData, msg.sender);
+    emit TransactionFulfilled(
+      txData.user,
+      txData.router,
+      txData.transactionId,
+      txData,
```

Comment:
Why are we passing in the whole `txData` on top of these properties? Does that not end up being redundant?

Reply:
Yes, but we have a limited number of event data fields we can `index`, so we choose the 3 most important. Indexed params make it faster to search through event logs